arangod/IResearch/IResearchRocksDBInvertedIndex.cpp

```diff
@@ -95,6 +95,15 @@ std::shared_ptr<Index> IResearchRocksDBInvertedIndexFactory::instantiate(
       definition, arangodb::StaticStrings::ObjectId);
   auto index = std::make_shared<IResearchRocksDBInvertedIndex>(
       id, collection, objectId, indexName);
+  auto& cf = _server.getFeature<ClusterFeature>();
+  if (cf.isEnabled()) {
+    std::string collectionName;
+    clusterCollectionName(
+        collection, clusterWideIndex ? nullptr : &cf.clusterInfo(),
+        index->Index::id().id(), false /*TODO meta.willIndexIdAttribute()*/,
+        collectionName);
+    index->setCollectionName(std::move(collectionName));
```

Comment:
Do we really need to request Agency each time? Maybe check if it is already set in meta and use that value?

Reply:
Done